src/cdk/tree/tree.ts

```diff
@@ -300,14 +302,19 @@ export class CdkTree<T> implements AfterContentChecked, CollectionViewer, OnDest
 @Directive({
   selector: 'cdk-tree-node',
   exportAs: 'cdkTreeNode',
-  host: {
-    '[attr.aria-expanded]': 'isExpanded',
-    '[attr.aria-level]': 'level + 1',
-    '[attr.role]': 'role',
-    'class': 'cdk-tree-node',
-  },
 })
-export class CdkTreeNode<T> implements FocusableOption, OnDestroy {
+export class CdkTreeNode<T> implements FocusableOption, OnDestroy, OnInit {
+  // TODO: mark as deprecated
+  /**
+   * The role of the tree node.
+   */
+  @Input() get role(): 'treeitem'|'group' { return 'treeitem'; }
+
+  set role(_role: 'treeitem'|'group') {
```

Comment:
Add `this.role = 'treeitem'` to the constructor so this is guaranteed to be initialized.